apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -15,13 +15,13 @@ open class DefaultApolloExtension(val project: Project)
   val services = mutableListOf<DefaultService>()
 
   /**
-   * compilationUnits is meant to be consumed by other gradle plugin.
+   * compilationUnit is meant to be consumed by other gradle plugin.
    * The apollo plugin will add the {@link CompilationUnit} as it creates them
    */
-  internal val compilationUnits = project.container(CompilationUnit::class.java)
+  internal val compilationUnit = project.container(CompilationUnit::class.java)
 
-  override fun onCompilationUnits(action: Action<CompilationUnit>) {
-    compilationUnits.all(action)
```

Comment:
What about keeping the old one and doing:

```
 fun onCompilationUnits(action: Action<CompilationUnit>) {
    throw Exception("please use onCompilationUnit {} instead")  
}
```

And then we remove that old version in a couple of months ?  This API is new from 1.3.0 so presumably not that many people are using it but that'll make a nicer deprecation path for the users that use it.